modules/calc/src/main/java/com/opengamma/strata/calc/runner/function/CalculationFunction.java

```diff
@@ -39,4 +61,32 @@
     return Optional.empty();
   }
 
+  /**
+   * Determines the market data requirements that function has to perform its calculations.
```

Comment:
Sounds a bit awkward, maybe "Determines the market data required by the function to perform its calculations".